### PR TITLE
Add spreadsheet modification for generator attributes

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/spreadsheet/utils/equipment-table-editors.js [hunk 1]
```diff
@@ -21,6 +21,55 @@ import {
     deepUpdateValue,
 } from './equipment-table-utils';
 
+export const StringField = forwardRef(
```
Comment (ayolab) [inline src/components/spreadsheet/utils/equipment-table-editors.js:24]: there is no need for this component, it has to be removed

**Review by ayolab — CHANGES_REQUESTED** [review]
there is no need for this StringField

File: src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    const { reset } = formMethods;
+    const clear = useCallback(() => {
+        reset(emptyFormData);
+    }, [reset]);
+
+    const [voltageLevelOptions, setVoltageLevelOptions] = useState([]);
+
+    useEffect(() => {
+        if (studyUuid && currentNode.id) {
+            fetchVoltageLevelsListInfos(studyUuid, currentNode.id).then(
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js:83]: can you treat error case please?

File: src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                    voltageLevelId: data.regulatingTerminalVlId,
+                }),
+            });
+        },
+        [reset]
+    );
+
+    useEffect(() => {
+        if (data) {
+            if (data.regulatingTerminalConnectableId !== ' ') {
```
Comment (ghelloumi) [inline src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js:115]: what if `data.regulatingTerminalConnectableId === ''` or 
`data.regulatingTerminalConnectableId === '   '` (n number of white spaces)

**Review by ghelloumi — APPROVED** [review]

**Review by AAJELLAL — COMMENTED** [review]
I found a weird behaviour while editing the line.
After changing the the voltage level and the equipment, the chosen voltage level is assigned to **regulatingTerminalVlId** . Based on the US description the voltage level should be changed but not **regulatingTerminalVlId**. To check when the PO to be sure.

**Review by AAJELLAL — APPROVED** [review]

**Comment by souissimai** [comment]
> I found a weird behaviour while editing the line. After changing the the voltage level and the equipment, the chosen voltage level is assigned to **regulatingTerminalVlId** . Based on the US description the voltage level should be changed but not **regulatingTerminalVlId**. To check when the PO to be sure.

we have **regulatingTerminalVlId** and **voltageLevelI** it's not the same

**Review by ayolab — APPROVED** [review]

**Pull request merged**